File: content/public/test/android/render_process_host_utils.cc
FASTVC-BENCH-CORPUS:chromium-main-v1-943b94ae-1c74-4335-94fa-ceb4d277cea8
// Copyright 2021 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "base/android/jni_android.h"
#include "content/public/browser/render_process_host.h"

// Must come after all headers that specialize FromJniType() / ToJniType().
#include "content/public/test/android/content_test_jni/RenderProcessHostUtils_jni.h"

namespace content {

jint JNI_RenderProcessHostUtils_GetCurrentRenderProcessCount(JNIEnv* env) {
  return RenderProcessHost::GetCurrentRenderProcessCountForTesting();
}

}  // namespace content

File: content/public/test/android/javatests/src/org/chromium/content_public/browser/test/util/RenderProcessHostUtils.java
FASTVC-BENCH-CORPUS:chromium-main-v1-943b94ae-1c74-4335-94fa-ceb4d277cea8
// Copyright 2021 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.content_public.browser.test.util;

import org.jni_zero.JNINamespace;
import org.jni_zero.NativeMethods;

import org.chromium.base.ThreadUtils;

/** Collection of test-only WebContents utilities. */
@JNINamespace("content")
public class RenderProcessHostUtils {
    private RenderProcessHostUtils() {}

    public static int getCurrentRenderProcessCount() {
        return ThreadUtils.runOnUiThreadBlocking(
                () -> {
                    return RenderProcessHostUtilsJni.get().getCurrentRenderProcessCount();
                });
    }

    @NativeMethods
    interface Natives {
        int getCurrentRenderProcessCount();
    }
}
